sbi/inference/snre/snre_base.py

```diff
@@ -90,17 +82,44 @@ def __init__(
             self._build_neural_net = utils.classifier_nn(model=classifier)
         else:
             self._build_neural_net = classifier
-        self._posterior = None
-        self._mcmc_method = mcmc_method
-        self._mcmc_parameters = mcmc_parameters
 
         # Ratio-based-specific summary_writer fields.
         self._summary.update({"mcmc_times": []})  # type: ignore
 
-    def __call__(
+    def add_data(
```

Comment:
it seems we have a separate `add_data` for every inference method, why is that?

Reply:
For SNLE and SNRE, I do not allow to pass the proposal. This is mostly to clearly show how the methods differ, and that SNRE and SNLE do not need to know about it.